coremltools/models/_infer_shapes_nn_mlmodel.py

```diff
@@ -457,6 +457,8 @@ def infer_shapes(nn_spec, input_spec, input_shape_dict = None):
         for inp in layer.input:
             assert inp in shape_dict, ('Input %s shape not cannot be determined' %(inp))
         layer_type = layer.WhichOneof('layer')
+        if layer_type == 'custom':
```

Comment:
I think it might be better to fill up the shape dictionary partially instead of not filling it at all. That is, let the code iterate over the layers and return when it encounters the first custom layer.

Reply:
Done